internal/app.go

```diff
@@ -207,18 +207,24 @@ func updateSecret(secretFile string, newContent []byte) (bool, error) {
 
 // Do an atomic write to the file which prevents race conditions for a reader.
 // Don't worry about writer synchronization as there is only one writer to these files.
-func safeWrite(secretFile string, newContent []byte) error {
-	tmp := secretFile + ".tmp"
-	// Remove a tmp file in case of an error; this fails in success case, but that can be ignored
-	defer os.Remove(tmp)
```

Comment:
I think we should still `defer os.Remove(tmp)`.
Otherwise, a failure to write this file will result into a leftover junk which prevents any further config updates.

Reply:
fixed in force push. note - the existence of the file won't prevent further updates. This is important because the 99% chance this function will fail is due to power failure - and it will be able to recover from that.